infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -783,34 +783,33 @@ private void loadTrimAddresses(SegmentHandle sh) throws IOException {
             pendingTrimSize = sh.getPendingTrimChannel().size();
         }
 
-        FileChannel fcTrimmed = getChannel(getTrimmedFilePath(sh.getFileName()), true);
-        if (fcTrimmed == null) {
-            return;
-        }
-        InputStream inputStream = Channels.newInputStream(fcTrimmed);
-
-        while (fcTrimmed.position() < trimmedSize) {
-            TrimEntry trimEntry = TrimEntry.parseDelimitedFrom(inputStream);
-            sh.getTrimmedAddresses().add(trimEntry.getAddress());
-        }
+        try (FileChannel fcTrimmed = getChannel(getTrimmedFilePath(sh.getFileName()), true)) {
+            try (InputStream inputStream = Channels.newInputStream(fcTrimmed)) {
 
-        inputStream.close();
-        fcTrimmed.close();
+                while (fcTrimmed.position() < trimmedSize) {
+                    TrimEntry trimEntry = TrimEntry.parseDelimitedFrom(inputStream);
+                    sh.getTrimmedAddresses().add(trimEntry.getAddress());
+                }
 
-        FileChannel fcPending = getChannel(getPendingTrimsFilePath(sh.getFileName()), true);
-        if (fcPending == null) {
-            return;
-        }
+                inputStream.close();
+                fcTrimmed.close();
 
-        inputStream = Channels.newInputStream(fcPending);
+                try (FileChannel fcPending =
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)